frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx

```diff
@@ -0,0 +1,141 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { RequestSizeInput, withHandlePromise } from '@console/internal/components/utils/index';
+import { StorageClassDropdown } from '@console/internal/components/utils/storage-class-dropdown';
+import {
+  createModalLauncher,
+  ModalTitle,
+  ModalBody,
+  ModalSubmitFooter,
+} from '@console/internal/components/factory';
+import './_add-capacity-modal.scss';
+import { k8sUpdate } from '@console/internal/module/k8s';
+
+export const AddCapacityModal = withHandlePromise((props: AddCapacityModalProps) => {
+  const { kind, obj, close, cancel } = props;
+  const dropdownUnits = {
+    TiB: 'TiB',
+    PiB: 'PiB',
+  };
+  const [requestSizeUnit, setRequestSizeUnit] = React.useState(dropdownUnits.TiB);
+  const [requestSizeValue, setRequestSizeValue] = React.useState('');
+  const [storageClass, setStorageClass] = React.useState('');
+  const [inProgress, setProgress] = React.useState(false);
+  const [errorMessage, setError] = React.useState('');
+  console.log(obj);
+  const getPresentValue = () => {
+    const currentValue = obj.spec.dataDeviceSet.capacity.split(' ');
+    const units = ['TiB', 'PiB'];
+    let value: any;
+    if (requestSizeUnit === units[0]) {
```

Comment:
Please check:
https://github.com/openshift/console/blob/master/frontend/public/components/utils/units.js#L322
https://github.com/openshift/console/blob/master/frontend/public/components/utils/units.js#L180

We might not require this function at all.

Reply:
ack